packages/react/src/components/Placeholder/placeholder-utils.tsx

```diff
@@ -0,0 +1,258 @@
+) => {
+  let result;
+  let phName = name.slice();
+
+  /**
+   * Process (SXA) dynamic placeholders
+   * Find and replace the matching dynamic placeholder e.g 'nameOfContainer-{*}' with the requested e.g. 'nameOfContainer-1'.
+   * For Metadata EditMode, we need to keep the raw placeholder name in place.
+   */
+  if (rendering?.placeholders) {
+    Object.keys(rendering.placeholders).forEach((placeholder) => {
+      const patternPlaceholder = isDynamicPlaceholder(placeholder)
+        ? getDynamicPlaceholderPattern(placeholder)
+        : null;
+
+      if (patternPlaceholder && patternPlaceholder.test(phName)) {
+        if (isEditing) {
+          phName = placeholder;
+        } else {
+          rendering.placeholders[phName] = rendering.placeholders[placeholder];
+          delete rendering.placeholders[placeholder];
+        }
+      }
+    });
+  }
+
+  if (rendering && rendering.placeholders && Object.keys(rendering.placeholders).length > 0) {
+    result = rendering.placeholders[phName];
+  } else {
+    result = null;
+  }
+
+  if (!result) {
+    console.warn(
+      `Placeholder '${phName}' was not found in the current rendering data`,
+      JSON.stringify(rendering, null, 2)
+    );
+
+    return [];
+  }
+
+  return result;
+};
+
+/**
+ * Get SXA specific params from Sitecore rendering params
+ * @param {ComponentRendering} rendering rendering object
+ * @returns {object} converted SXA params
+ */
+export const getSXAParams = (rendering: ComponentRendering) => {
+  if (!rendering.params) return {};
+
+  const { GridParameters, Styles } = rendering.params;
+
+  return (
+    (GridParameters || Styles) && {
+      styles: `${GridParameters || ''} ${Styles || ''}`,
+    }
+  );
+};
+
+/**
+ * Renders the placeholder when it is empty. The required CSS styles are applied to the placeholder in edit mode.
+ * @param {React.ReactNode | React.ReactElement[]} node react node
+ * @returns react node
+ */
+export const renderEmptyPlaceholder = (node: React.ReactNode | React.ReactElement[]) => {
+  return <div className="sc-jss-empty-placeholder">{node}</div>;
+};
+
+/**
+ * Wraps rendered component(s) with an error boundary.
+ * @param {React.ReactElement} rendered React elements to be wrapped
+ * @param {PlaceholderProps} placeholderProps props of placeholder under which the component is being rendered
+ * @param {string} renderingKey unique key for the error boundary
+ * @param {boolean} isDynamic whether error boundary wraps any lazy components
+ * @param {object} [extraProps] extra non-rendered component props to be passed to the error boundary
+ * @returns {React.ReactElement} wrapped component
+ */
+export const wrapErrorBoundary = (
+  rendered: React.ReactElement<{ [attr: string]: unknown }>,
+  placeholderProps: PlaceholderProps,
+  renderingKey: string,
+  isDynamic: boolean = false
+) => {
+  const disableSuspense = placeholderProps.disableSuspense || false;
+  const serializableProps = nonSerializedProps.reduce(
```

Comment:
This nonSerialized props hack is strange from this helper perspective for two reasons:
- it is applied regardless it is client or server context
- ErrorBoundary component doesn't expect arbitrary key value props and don't need all this stuff from placeholderProps

Reply:
Adjusted function to be more generic

Reply:
and thus no need for all this pick serializable prop helpers